File: app/src/main/java/com/blog/demo/image/animation/TransitionAnimationActivity.java
package com.blog.demo.image.animation;

import android.app.Activity;
import android.app.ActivityOptions;
import android.content.Intent;
import android.os.Bundle;
import android.transition.Slide;
import android.transition.Transition;
import android.transition.TransitionSet;
import android.view.Gravity;
import android.view.View;

import androidx.annotation.Nullable;

import com.blog.demo.R;

public class TransitionAnimationActivity extends Activity implements View.OnClickListener {

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_image_animation_transition);

        findViewById(R.id.iv_link).setOnClickListener(this);
    }

    @Override
    public void onClick(View view) {
        if (view.getId() == R.id.iv_link) {
            Intent intent = new Intent(this, TransitionAnimationResultActivity.class);

            Transition transition = new TransitionSet()
                    .addTransition(new Slide(Gravity.LEFT).addTarget(R.id.iv_link).addTarget(R.id.iv_pyq))
                    .addTransition(new Slide(Gravity.RIGHT).addTarget(R.id.iv_qq).addTarget(R.id.iv_wx))
                    .setDuration(1000);

            getWindow().setExitTransition(transition);
            Bundle bundle = ActivityOptions.makeSceneTransitionAnimation(this).toBundle();

            startActivity(intent, bundle);
        }
    }
}

File: app/src/main/java/com/blog/demo/image/animation/TransitionAnimationActivity.kt
package com.blog.demo.image.animation

import android.app.Activity
import android.app.ActivityOptions
import android.content.Intent
import android.os.Bundle
import android.transition.Slide
import android.transition.TransitionSet
import android.view.Gravity
import android.view.View
import android.widget.ImageView
import com.blog.demo.R

open class TransitionAnimationActivity : Activity(), View.OnClickListener {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        setContentView(R.layout.activity_image_animation_transition)
        findViewById<ImageView>(R.id.iv_link).setOnClickListener(this)
    }

    override fun onClick(view: View) {
        if (view.id == R.id.iv_link) {
            val intent = Intent(this, TransitionAnimationResultActivity::class.java)
            window.exitTransition = TransitionSet()
                .addTransition(Slide(Gravity.LEFT).addTarget(R.id.iv_link).addTarget(R.id.iv_pyq))
                .addTransition(Slide(Gravity.RIGHT).addTarget(R.id.iv_qq).addTarget(R.id.iv_wx))
                .setDuration(1000)

            val bundle = ActivityOptions.makeSceneTransitionAnimation(this).toBundle()
            startActivity(intent, bundle)
        }
    }

}